eodatasets3/wagl_md_only.py

```diff
@@ -1,60 +1,92 @@
 #!/usr/bin/env python
 
-from datetime import datetime
 from posixpath import join as ppjoin, basename as pbasename
 from pathlib import Path
 import h5py
 from rasterio.crs import CRS
 from affine import Affine
-from eodatasets3 import DatasetAssembler, images
+from eodatasets3 import DatasetAssembler, images, utils
+import eodatasets3.wagl
 from wagl.hdf5 import find
+from datetime import datetime
+from rasterio.enums import Resampling
+import os
 
-
-INDIR = Path("/g/data/v10/testing_ground/jps547/test-jq-indexing/workdir/batchid-6f608760a3/jobid-635fa0/LC08_L1GT_088078_20151206_20170401_01_T2.tar.ARD/LC80880782015340LGN01")
-WAGL_FNAME = Path("LC80880782015340LGN01.wagl.h5")
-GROUP_PATH = "/LC80880782015340LGN01/RES-GROUP-1/STANDARDISED-PRODUCTS"
+INDIR = Path("/g/data/up71/projects/index-testing-wagl/wagl/workdir/batchid-48b378b0f0/jobid-c59136/LC08_L1TP_099080_20160613_20180203_01_T1.tar.ARD/LC80990802016165LGN02")
+WAGL_FNAME = Path("LC80990802016165LGN02.wagl.h5")
+GROUP_PATH = "/LC80990802016165LGN02/RES-GROUP-1/STANDARDISED-PRODUCTS"
 GDAL_H5_FMT = 'HDF5:"{filename}":/{dataset_pathname}'
 
-
-def package(outdir):
+def package_non_standard(outdir, granule):
     """
-    Quick toy yaml creator for the ard pipeline.
+    yaml creator for the ard pipeline.
     Purely for demonstration purposes only.
     Can easily be expanded to include other datasets.
 
     A lot of the metadata such as date, can be extracted from the yaml
     doc contained withing the HDF5 file at the following path:
 
     [/<granule_id>/METADATA/CURRENT]
+    """
 
-    
+    #out_fname = outdir.joinpath(granule.name+'.yaml')
     """
-    out_fname = INDIR.joinpath('LC80880782015340LGN01.yaml')
-    with DatasetAssembler(metadata_path=out_fname, naming_conventions='dea') as da:
-        da.platform = 'landsat8'
+    with DatasetAssembler(Path(outdir), naming_conventions='dea') as da:
+
+        if granule.fmask_image:
+            with eodatasets3.wagl.do(f"Writing fmask from {granule.fmask_image} "):
+                da.write_measurement(
```

Comment:
Don't worry about using `write_measurement` for this feature. Stick with `note_measurement`.